### ISSUE TITLE
Indent rule's flatTernaryExpressions option isn't working

### ISSUE THREAD

**Issue by Sawtaytoes** [eslint/eslint#12968]
**Tell us about your environment**

**Node version:** v10.19.0
**npm version:** v6.13.4
**Local ESLint version:** v6.8.0
**Global ESLint version:** Not found

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
	'env': {
		'browser': true,
		'es6': true
	},
	'extends': 'eslint:recommended',
	'globals': {
		'Atomics': 'readonly',
		'SharedArrayBuffer': 'readonly'
	},
	'parserOptions': {
		'ecmaFeatures': {
			'jsx': true
		},
		'ecmaVersion': 2018,
		'sourceType': 'module'
	},
	'plugins': ['react'],
	'rules': {
		'indent': [
			'error',
			'tab',
			{
				flatTernaryExpressions: true,
				MemberExpression: 0,
				SwitchCase: 1,
			},
		],
	},
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let func

func = ({
	hello,
	someone,
}) =>
	someone
	? someone
	: hello

func = ({
	hello,
	someone,
}) => (
	someone
	? someone
	: hello
)

const test = (
	func()
	? (
		blah
		? 'phil'
		: 'kevin'
	)
	: 'jacob'
)

func
test
```

```bash
yarn eslint index.js
```

**What did you expect to happen?**
I expected it allow ternaries to be flat; thus, not requiring a tab before the `?` and `:`.

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
  17:1  error  Expected indentation of 1 tab but found 0   indent
  65:1  error  Expected indentation of 2 tabs but found 1  indent
  66:1  error  Expected indentation of 2 tabs but found 1  indent
  73:1  error  Expected indentation of 2 tabs but found 1  indent
  74:1  error  Expected indentation of 2 tabs but found 1  indent
  81:1  error  Expected indentation of 2 tabs but found 1  indent
  82:1  error  Expected indentation of 3 tabs but found 2  indent
  83:1  error  Expected indentation of 4 tabs but found 2  indent
  84:1  error  Expected indentation of 4 tabs but found 2  indent
  85:1  error  Expected indentation of 2 tabs but found 1  indent
  86:1  error  Expected indentation of 2 tabs but found 1  indent

✖ 11 problems (11 errors, 0 warnings)
  11 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Are you willing to submit a pull request to fix this bug?**
Yes.

**Comment by Sawtaytoes** [comment]
I might've misunderstood how `flatTernaryExpression` is supposed to work so I provided a PR for the change instead #12977.

**Comment by kaicataldo** [comment]
Apologies for the delay. Can you update this proposal to match the PR? It's hard to understand the full proposal with them being in different states.

**Comment by Sawtaytoes** [comment]
I think there is a bit more work I need to do to get this working properly because I think there's a bug in `indent` when your value is set to `0`.
